rules/no-mod-iter-var-for-loop.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * @fileoverview Flag for loops which modify their iteration variable in their body
+ * @author Nicolas Feignon <nfeignon@gmail.com>
+ */
+
+'use strict';
+
+module.exports = {
+
+    meta: {
+
+        docs: {
+            recommended: true,
+            type: 'error',
+            description: 'Flag for loops which modify their iteration variable in their body'
+        },
+
+        schema: []
+
+    },
+
+    create: function (context) {
+
+        function inspectLoopStatement(emitted) {
+            var node = emitted.node;
+            if (!emitted.exit) {
+                if (!node.init || node.init.type !== 'AssignmentExpression')
+                    return;
+                var iterationVariable = node.init.left.name;
+
+                for (let expr of node.body.body) {
+                    if (expr.type === 'ExpressionStatement') {
+
+                        if (expr.expression.type === 'AssignmentExpression' && expr.expression.left.type === 'Identifier') {
+                            var name = expr.expression.left.name;
+                        } else if (expr.expression.type === 'UpdateExpression' && expr.expression.argument.type === 'Identifier') {
+                            var name = expr.expression.argument.name;
+                        } else
+                            continue;
+
+                        if (name === iterationVariable) {
+                            context.report({
+                                node: expr,
+                                message: 'Iterator variable modified in for loop'
+                            });
+                        }
```

Comment:
Please refactor the code to avoid the [Pyramid of Doom](http://i.imgur.com/MByWioX.png) as much as you can :P This makes code unmanageable down the line.
You could take the blocks out. For eg-
Instead of 
```js
if (!emitted.exit) {
   // logic
}
``` 

You could do
```js
if (emitted.exit) { return; }

// logic
```

At some places it may be outright necessary (in which case its fine), but avoid indentation depth wherever you can